hphp/hack/src/hh_codesynthesis/agentGraphGenerator.py

```diff
@@ -175,33 +190,13 @@ def validate(
             for x in depends_on:
                 out_degrees[x] += 1
 
-        for node, degree in enumerate(in_degrees):
-            agent_profile = self.solving_context.infra_agent_profile
-            if node in self.product_agents:
-                agent_profile = self.solving_context.product_agent_profile
-
-            if (
-                degree < agent_profile["in_degree"][0]
-                or degree > agent_profile["in_degree"][1]
-            ):
-                raise RuntimeError(
-                    "node: {0}, degree: {1}, models: {2}".format(
-                        node, degree, self._raw_model
-                    )
-                )
-                return False
-
-        for node, degree in enumerate(out_degrees):
-            agent_profile = self.solving_context.infra_agent_profile
-            if node in self.product_agents:
-                agent_profile = self.solving_context.product_agent_profile
-
-            if (
-                degree < agent_profile["out_degree"][0]
-                or degree > agent_profile["out_degree"][1]
-            ):
-                return False
+        # Validate the degrees are in the range specified by the profile.
+        if self.validate_range_in_profile(in_degrees, "in_degree") is False:
```

Comment:
nit: use if not

Reply:
We are using "assert" inside the function, so just invoke the function here.